decisions/websre/0005-web-sre-kubernetes-secrets-pattern.md

```diff
@@ -0,0 +1,40 @@
+# 1. web-sre-kubernetes-secrets-pattern
+
+Date: 2021-06-09
+
+## Status
+
+Accepted
+
+## Context
+
+Secrets are inconsistently managed across our various clusters.  This makes it hard to trace where secrets come from and update them.  It also increases our exposure to security issues, as our each tool used can become compromised independently.  Current known secret management includes git-crypt repos, with sops, that apply kubernetes secrets direct to the cluster; sealed secrets; and external secrets pointing at cloud provider secret storage.  There are a few other methods that treat things that should be secrets incorrectly. For example, baking the 'secrets' into the docker image directly. We will treat those situations with more urgency, viewing them as explicitly a bad state to be in.
+
+The goal of this document is to describe the happy path for passing secrets to kubernetes based applications.
```

Comment:
It would be more appealing if we separates "we will use the cloud provider's secret manager" and "for kubernetes, we will use the External Secrets module", so that in cases where we aren't using kubernetes (for example, lambda-only deployments) the "secret manager" recommendation remains valid.

Reply:
I think this pretty clearly states the goal of this ADR is for kubernetes-based applications though, which starts with the cloud provider secret manager. What wording do you feel needs to be added to make it more explicit this doesn't necessarily cover non-kubernetes-based applications?

Reply:
One habit I've seen in other ADR processes is the "Out of Scope" section of the document, but in lieu of that, "Secrets management for non-Kubernetes is out of scope for this ADR" would be fine.